frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/utils.ts

```diff
@@ -55,3 +89,218 @@ export function checkIfCancelled(datasetStatus: DatasetUploadStatus): boolean {
     datasetStatus.upload_status === UPLOAD_STATUS.CANCELED
   );
 }
+
+export function useCheckCollectionPopulated({
+  invalidateCollectionQuery,
+  isNamePopulated,
+  upload_progress,
+}: {
+  invalidateCollectionQuery: () => void;
+  isNamePopulated: boolean;
+  upload_progress: number;
+}) {
+  useCheckCollection({
+    invalidateCollectionQuery,
+    shouldFetch: upload_progress === 1 && !isNamePopulated,
+  });
+}
+
+type FORMAT_KEYS =
+  | "conversion_loom_status"
+  | "conversion_anndata_status"
+  | "conversion_cxg_status"
+  | "conversion_rds_status";
+
+const CONVERSION_STATUS_FORMAT_KEYS = [
+  "conversion_loom_status",
+  "conversion_anndata_status",
+  "conversion_cxg_status",
+  "conversion_rds_status",
+] as FORMAT_KEYS[];
+
+export function useCheckCollectionFormatsPopulated({
+  invalidateCollectionQuery,
+  dataset,
+  datasetUploadStatus,
+}: {
+  invalidateCollectionQuery: () => void;
+  dataset: Dataset;
+  datasetUploadStatus: DatasetUploadStatus;
+}) {
+  let numOfConvertedFormats = 0;
+
+  for (const key of CONVERSION_STATUS_FORMAT_KEYS) {
+    if (datasetUploadStatus[key] !== CONVERSION_STATUS.CONVERTED) continue;
+
+    numOfConvertedFormats += 1;
+  }
+
+  let numOfAvailableFormats = 0;
+
+  if (dataset.dataset_deployments.length > 0) {
+    numOfAvailableFormats += 1;
+  }
+
+  numOfAvailableFormats = numOfAvailableFormats + dataset.dataset_assets.length;
```

Comment:
Yeah! Essentially we know available formats only exist in `dataset.dataset_deployments` and `dataset.dataset_assets`. Where CXG live in `dataset_deployments` and other formats in `dataset_assets`

So if the status tells us CXG, H5AD, and LOOM are converted, but we only see CXG in `dataset_deployments` and H5AD in `dataset_assets`, then the count is `2` and we know we need to keep polling `collection` endpoint until we get `LOOM` in `dataset_assets`, or when the count gets to `3`

And because each format only has one corresponding format, instead of tracking what format is actually available, we can just track the counts!